XdfStreamer/xdfstreamer.cpp

```diff
@@ -180,6 +180,21 @@ lsl::stream_info XdfStreamer::initializeLslStreamsForXdfData(const int samplingR
 
     lsl::stream_info info(streamName, streamType, channelCount, (double)samplingRate, channelFormat, "RT_Sender_SimulationPC");
 
+    // Reading channel names from the xdf file and setting for the stream!
+    std::vector<std::map<std::string, std::string> > channels = this->xdf->streams[this->stream_idx].info.channels;
+
+    lsl::xml_element chns = info.desc().append_child("channels");
+    for(const std::map<std::string, std::string> &item : channels) {
+
+        std::vector<std::pair<std::string, std::string>> ch_info;
+        for(auto &[key, val] : item) {
+            ch_info.push_back(std::make_pair(key, val));
+        }
+        chns.append_child("channel")
```

Comment:
I think you're assuming that each channel will have 2 elements, probably like "label" and "unit". But this isn't necessarily true. It's probably better to first make a channel object, then do a loop based on how many things were pushed into ch_info:

```
lsl::xml_element chn = chns.append_child("channel");
for (const auto& kv_pair: ch_info) {
    chn.append_child_value(kv_pair.first, kv_pair.second);
}
```

Of course, the optimal solution would be to have a full traversal of the xdf_stream.info with some generic + recursive functions. But I know that's a lot of work, and getting 80% of the way there for 10% of the work is a good tradeoff here.

Reply:
@cboulay you are right the assumption may not be correct. I will fix and re-submitted again. I am also working ob the multi-stream version of it. I am testing now, hopefully will submit soon.